### PR TITLE
Add build plugin to webpack build.

### PR DESCRIPTION
**Pull request by yoannmoinet**
**What's the problem this PR addresses?**

Gather more data from within the webpack build with the help of [Datadog's build-plugin](https://github.com/DataDog/build-plugin).

Later on, we'll create a public dashboard in order to show how our build is.

You will need to add [`DD_API_KEY`](https://app.datadoghq.eu/account/settings#api) in the secrets, in order to submit the metrics to Yarn's org.

**How did you fix it?**

Added the [build-plugin](https://github.com/DataDog/build-plugin) and configured it.

Here's some captures of what we can have:

|Logs output | Dashboard |
|:-----------:|:----------:|
|Screenshot 2020-09-22 at 16 22 51|image|

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [ ] I have verified that all automated PR checks pass.

### CODE REVIEW

File: packages/yarnpkg-builder/package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "webpack-sources": "^1.4.3"
   },
   "devDependencies": {
+    "@datadog/build-plugin": "0.3.3",
```
Comment (merceyz) [inline packages/yarnpkg-builder/package.json:27]: Needs to be a normal dependency, or we need to modify the builder so we only have this enabled when run in the berry repo

**Comment by arcanis** [comment]
@merceyz remarked that third-party people using the builder to build their own plugins may not want to send data to Datadog. I don't know if the token having to be in the environment is enough or not - would it be worth making this optional somehow? Problem is, I don't think the builder is very configurable at the moment... 🤔

**Comment by yoannmoinet** [comment]
@arcanis we could use a "not so common" env var name so if it's not present, nothing gets send.
Something like `YARN_DD_API_KEY_BUILD_PLUGIN`, so you would be very unlucky to have this specific (and correct) var in your environment.

If you were very unlucky, had it in your env, and be it correct, it would send data to your org, not yarn's org anyway.

**Comment by yoannmoinet** [comment]
Apparently, we can't install GitHub packages anonymously.

So I have to work a bit on the plugin first to make it open-source ready and have it in NPM's registry.

If that's ok to keep this PR open for the next 2 days or so.

**Comment by yoannmoinet** [comment]
All right, the package is now public and on NPM's registry.
We're good to go.
For `Integration / Testing chores (pull_request)`, I'm not sure which packages should be bumped, I thought only `builder` and `cli` should be bumped.

**Review by merceyz — CHANGES_REQUESTED** [review]
> For Integration / Testing chores (pull_request), I'm not sure which packages should be bumped, I thought only builder and cli should be bumped.

You can set all of them to declined.

I'd like to only have this applied when run within our repo, we can add a very specific env variable check that only triggers on our CI until we've added support for custom webpack configs to `@yarnpkg/builder`.

**Review by merceyz — APPROVED** [review]
I've made it only be setup and installed when in our repo (sort of), so i'm happy at least

**Comment by yoannmoinet** [comment]
Thanks @merceyz !
I think we can also remove the `disabled: !process.env. BUILD_MONITORING_ENABLED` part from the config.

File: packages/yarnpkg-builder/sources/tools/makeConfig.ts [hunk 2]
```diff
@@ -79,6 +79,19 @@ export const makeConfig = (config: webpack.Configuration): webpack.Configuration
   },
 
   plugins: [
+    process.env.BUILD_MONITORING_ENABLED && new (require(`@datadog/build-plugin/dist/webpack`).BuildPlugin)({
```
Comment (arcanis) [inline packages/yarnpkg-builder/sources/tools/makeConfig.ts:82]: Might be worth using something like:

```ts
function getBuildPlugin() {
  if (!test)
    return [];
  const BuildPlugin = require('...')
  return [new BuildPlugin()];
}
plugins: [
  ...getBuildPlugins(),
  new OtherPlugin1(),
  new OtherPlugin2(),
]
```

This way we don't need to type or filter the array

File: packages/yarnpkg-builder/sources/tools/makeConfig.ts [hunk 3]
```diff
@@ -31,6 +31,28 @@ export function findTsconfig() {
...
+  const BuildPlugin = require(`@datadog/build-plugin/dist/webpack`).BuildPlugin;
+  return [new BuildPlugin({
+    context: npath.join(process.cwd(), `../..`),
+    datadog: {
+      apiKey: process.env.DD_API_KEY,
+      prefix: `webpack`,
+      tags: [
+        `branchname:${process.env.CIRCLE_BRANCH || `branch`}`,
+        `sha:${process.env.CIRCLE_SHA1 || `local`}`,
+        `jobname:${process.env.CIRCLE_JOB || `job`}`,
```
Comment (merceyz) [inline packages/yarnpkg-builder/sources/tools/makeConfig.ts:49]:
```suggestion
        `branchname:${process.env.GITHUB_REF || `branch`}`,
        `sha:${process.env.GITHUB_SHA || `local`}`,
        `jobname:${process.env.GITHUB_EVENT_NAME || `job`}`,
```
https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables

Comment (yoannmoinet) [reply packages/yarnpkg-builder/sources/tools/makeConfig.ts:49]: I don't know why I thought this build was running on CircleCI 🤦

**Comment by arcanis** [comment]
Let the build data flow

**Pull request merged**